### PR TITLE
[Steam] Fix extractor

### LINKED ISSUE #1992: [Steam] Unable to extract
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2021.12.01**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

_No response_

### Description

Steam doesn't work eqeqe

### Verbose log

```shell
$ yt-dlp -v https://store.steampowered.com/app/105600/Terraria/
[debug] Command-line config: ['-v', 'https://store.steampowered.com/app/105600/Terraria/']
[debug] User config file: /home/pogaca/.config/yt-dlp/config
[debug] User config: ['-f', 'mkv', '-f', 'bestvideo[height<=1080]+bestaudio/best/standard', '-o', '~/Downloads/%(title).200s - %(id)s.%(ext)s']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2021.12.01 [91f071af6]
[debug] Lazy loading extractors is disabled
[debug] Python version 3.9.9 (CPython 64bit) - Linux-5.15.7-arch1-1-x86_64-with-glibc2.33
[debug] exe versions: ffmpeg 4.4.1 (setts), ffprobe 4.4.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [Steam] Extracting URL: https://store.steampowered.com/app/105600/Terraria/
[Steam] 105600: Downloading webpage
ERROR: [Steam] Unable to extract flash vars; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/yt_dlp/extractor/common.py", line 599, in extract
    ie_result = self._real_extract(url)
  File "/usr/lib/python3.9/site-packages/yt_dlp/extractor/steam.py", line 88, in _real_extract
    flash_vars = self._parse_json(self._search_regex(
  File "/usr/lib/python3.9/site-packages/yt_dlp/extractor/common.py", line 1173, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
yt_dlp.utils.RegexNotFoundError: Unable to extract flash vars; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/yt_dlp/extractor/common.py", line 599, in extract
    ie_result = self._real_extract(url)
  File "/usr/lib/python3.9/site-packages/yt_dlp/extractor/steam.py", line 88, in _real_extract
    flash_vars = self._parse_json(self._search_regex(
  File "/usr/lib/python3.9/site-packages/yt_dlp/extractor/common.py", line 1173, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
yt_dlp.utils.RegexNotFoundError: Unable to extract flash vars; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/yt_dlp/YoutubeDL.py", line 1331, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/lib/python3.9/site-packages/yt_dlp/YoutubeDL.py", line 1400, in __extract_info
    ie_result = ie.extract(url)
  File "/usr/lib/python3.9/site-packages/yt_dlp/extractor/common.py", line 625, in extract
    raise type(e)(e.msg, **kwargs)
yt_dlp.utils.RegexNotFoundError: [Steam] Unable to extract flash vars; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
```

### PR DESCRIPTION
**Pull request by u-spec-png**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Closes #1992

### CODE REVIEW

File: yt_dlp/extractor/steam.py [hunk 1]
```diff
@@ -53,96 +53,62 @@ class SteamIE(InfoExtractor):
...
-        fileID = m.group('fileID')
-        if fileID:
-            videourl = url
-            playlist_id = fileID
-        else:
-            gameID = m.group('gameID')
-            playlist_id = gameID
-            videourl = self._VIDEO_PAGE_TEMPLATE % playlist_id
+        gameID = m.group('gameID')
+        playlist_id = gameID
```
Comment (pukkandan) [inline yt_dlp/extractor/steam.py:70]:
```suggestion
        playlist_id = self._match_valid_url(url).group('gameID')
```

what ae the other groups in the VALID_URL for?

File: yt_dlp/extractor/steam.py [hunk 2]
```diff
@@ -53,96 +53,62 @@ class SteamIE(InfoExtractor):
...
+        videourl = self._VIDEO_PAGE_TEMPLATE % playlist_id
```
Comment (pukkandan) [inline yt_dlp/extractor/steam.py:71]:
```suggestion
        video_url = self._VIDEO_PAGE_TEMPLATE % playlist_id
```

File: yt_dlp/extractor/steam.py [hunk 3]
```diff
@@ -53,96 +53,62 @@ class SteamIE(InfoExtractor):
...
+                for quality in ('', '-hd'):
+                    for ext in ('webm', 'mp4'):
+                        video_url = movie.get('data-%s%s-source' % (ext, quality))
+                        if video_url:
+                            formats.append({
+                                'format_id': ext + quality,
+                                'url': video_url,
+                            })
+            if not formats and not self.get_param('ignore_no_formats'):
+                continue
```
Comment (pukkandan) [inline yt_dlp/extractor/steam.py:109]: not needed

File: yt_dlp/extractor/steam.py [hunk 4]
```diff
@@ -53,96 +53,62 @@ class SteamIE(InfoExtractor):
...
+            entry['formats'] = formats
```
Comment (pukkandan) [inline yt_dlp/extractor/steam.py:110]:
```suggestion
            self._sort_formats(formats)
            entry['formats'] = formats
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/steam.py [hunk 5]
```diff
@@ -18,30 +17,31 @@ class SteamIE(InfoExtractor):
             (?P<urltype>video|app)/ #If the page is only for videos or for a game
             (?P<gameID>\d+)/?
             (?P<videoID>\d*)(?P<extra>\??) # For urltype == video we sometimes get the videoID
-        |
-        https?://(?:www\.)?steamcommunity\.com/sharedfiles/filedetails/\?id=(?P<fileID>[0-9]+)
```
[the remark this answers is not in the document]
Comment (pukkandan) [reply yt_dlp/extractor/steam.py]: I was not asking to blindly add it back. I see that you have removed the test for this...

Please communicate. Don't make me have to guess your intend

[the remark this answers is not in the document]
Comment (u-spec-png) [reply yt_dlp/extractor/steam.py]: Test was dead and couldn't find any link that sent me to a videos, so I thought that links were no longer valid.

[the remark this answers is not in the document]
Comment (u-spec-png) [reply yt_dlp/extractor/steam.py]: They still work but they're all YouTube embedded.

**Pull request merged**